### PR TITLE
change enable modal to toggle between enable and disable

### PR DESCRIPTION
**Pull request by danoswaltCL**
This actually changes the "enable" modal into a template that is used for both enable and disable, only difference was really the content of the message and some css that is easily configurable in the commonModalDialog we already have.

@Yagnik56 i also consolidated all the actions / effects from "enable" to just "updateFlagStatus".

### CODE REVIEW

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-details-page/feature-flag-details-page-content/feature-flag-overview-details-section-card/feature-flag-overview-details-section-card.component.ts [hunk 1]
```diff
@@ -54,18 +60,33 @@ export class FeatureFlagOverviewDetailsSectionCardComponent {
...
+    const dialogRef = this.dialogService.openDisableFeatureFlagConfirmModel();
+
+    dialogRef.afterClosed().subscribe(() => {
+      this.changeDetectorRef.detectChanges();
+    });
+  }
+
+  listenForClosedModal(dialogRef: MatDialogRef<UpdateFlagStatusConfirmationModalComponent>): void {
+    dialogRef.afterClosed().subscribe(() => {
+      this.changeDetectorRef.detectChanges();
```
Comment (Yagnik56) [inline frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-details-page/feature-flag-details-page-content/feature-flag-overview-details-section-card/feature-flag-overview-details-section-card.component.ts:88]: I didn't understand the need for this `this.changeDetectorRef.detectChanges();` and why listen to it separately and in the model sub. can you share why?

**Review by Yagnik56 — DISMISSED** [review]
everything else looks good so I am approving this PR, but didn't understand the flow in the overview section card for the modal so would be nice if you could explain it before we merge this.

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-details-page/feature-flag-details-page-content/feature-flag-overview-details-section-card/feature-flag-overview-details-section-card.component.ts (hunk 1 above)
Comment (danoswaltCL) [reply frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-details-page/feature-flag-details-page-content/feature-flag-overview-details-section-card/feature-flag-overview-details-section-card.component.ts:88]: hmI I think that looks like something that I left in as I was trying for debugging getting the chip-status to update after enable, but it didn't work. I'll remove them.

**Review by Yagnik56 — APPROVED** [review]
It looks good, I checked before and it worked fine, so after the last change @danoswaltCL you made it function the same, we can merge this.

**Pull request merged**